README.md

```diff
@@ -310,40 +348,113 @@ aws ec2 describe-instances --region eu-west-1
-
-*Note: if you want to run the example using the AWS instances, be aware that
+*Note: if you want to run the examples using the AWS instances, be aware that
 this has a cost. You can change the value of
 `"max_bid_price_in_dollars_per_hour": N` in `plz.config.json` to any value
-you like. The example takes around 5 minutes to run.
-The value in the provided file is 0.5 dollars/hour. See the following note as
-well.*
+you like. Examples takes around 5 minutes to run.
+The value in the provided configs range from 0.5 dollars/hour to 2 dollars/hour
+(for GPU-powered machines). See the following note as well.*
 
 *Note: unless you add `"instance_max_uptime_in_minutes": null,` to your
-`plz.config.json`, the instance terminates after 60 minutes.  That's on
-purpose, in case you're just trying the tool and something doesn't go well
-(like, there's a power cut). You can always use `plz list` and `plz kill`
-before leaving your computer, as to make sure that there no instances remaining.
-For maximum assurance, you can check your instances in the AWS console.*
+`plz.config.json`, all AWS instances you start terminate after 60 minutes.
+That's on purpose, in case you're just trying the tool and something doesn't
+go well (like, there's a power cut). You can always use `plz list` and
+`plz kill` before leaving your computer, as to make sure that there no
+instances remaining. For maximum assurance, you can check your instances in
+the AWS console.*
+
+
+### Pytorch
+
+In the directory `plz/examples/pytorch` there's a full-fledged example that
+shows how to do the classical training for character recognition using
+LeNets, using a subset of the well-known MNIST dataset. There's a simple
+example using only python, described in the section below.
+
+Anything related to `plz` is in `main.py`. In fact the most relevant lines
+are the following ones:
+
+```
+def get_from_plz_config(key: str, non_plz_value: T) -> T:
+    configuration_file = os.environ.get('CONFIGURATION_FILE', None)
+    if configuration_file is not None:
+        with open(configuration_file) as c:
+            config = json.load(c)
+        return config[key]
+    else:
+        return non_plz_value
+[...]
+    input_directory = get_from_plz_config(
+        'input_directory', os.path.join('..', 'data'))
+    output_directory = get_from_plz_config('output_directory', 'models')
+    parameters = get_from_plz_config('parameters', DEFAULT_PARAMETERS)
+    measures_directory = get_from_plz_config('measures_directory', 'measures')
+    summary_measures_path = get_from_plz_config(
+        'summary_measures_path',
+        os.path.join('measures', 'summary'))
+```
+This shows how to get the input data and parameters that `plz` uploads
+for you. There's a configuration file whose name comes in the environment
+variable `CONFIGURATION_FILE`. If that variable is present, you're running
+with `plz`, and you can read and parse the file as a json object. The object
+has the following keys:
```

Comment:
Is this just convention, or is it enforced?

Reply:
I guess, my point is, this isn't enforced, but we are getting very specific in explaining the keys etc, while it could really be anything. Had me confused for a bit.

Reply:
I'm struggling to understand what makes you think the key names are just conventions... plz needs to know where your input files are. They are in `../data/mnist` in this case, and plz uploads those because this path is the value for the key `input` in `plz.config.json`. Then it needs to put them somewhere for you to read them, and it puts them in `config['input_directory']`. If this were just convention how do you think this works? It uploads all files under all keys in `plz.config.json`? All keys in the config correspond to input directories?

Whatever reading you made someone else could make it, so I'm interested about how you've read it, as to make the README less confusing :)

Reply:
Sorry, I'm an idiot and I have the flu... I was thinking that, if I have a key called 'temp repo' and I read that json and start writing files to the temp repo, I am not sure if plz will stop me. But obviously at least some of the keys are meaningful.